common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -639,7 +624,7 @@ public void onData(String streamId, ByteBuffer buf) throws IOException {
           // written to disk due to this reason. We thus decide to optimize for server
           // throughput and memory usage.
           if (deferredBufs == null) {
-            deferredBufs = new LinkedList<>();
+            deferredBufs = new ArrayList<>();
```

Comment:
Why changed to `ArrayList`?

Reply:
Quoting @mridulm 
"make this ArrayList instead - LinkedList has bad perf charactestics in comparison, and is preferable only when we have removes."

Reply:
Could you fix Java Linter Error? GitHub Action is complaining for the following.
```
src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java:[33,8] (imports) UnusedImports: Unused import - java.util.LinkedList.
```

Reply:
Sure, fixed it.